### PR TITLE
Version Compatibility Check #154

### LINKED ISSUE #154: Advertise version
Advertise `xud` `version` on handshaking.
When received, verify that it exceeds `MIN_VERSION`. If not, disconnect from peer.

### PR DESCRIPTION
**Pull request by rsercano**
resolves #154, please ignore #757

### CODE REVIEW

File: lib/Xud.ts [hunk 1]
```diff
@@ -87,7 +89,7 @@ class Xud extends EventEmitter {
         initPromises.push(this.raidenClient.init());
       }
 
-      this.pool = new Pool(this.config.p2p, loggers.p2p, this.db.models);
+      this.pool = new Pool(this.config.p2p, loggers.p2p, this.db.models, minCompatibleVersion);
```
Comment (moshababo) [inline lib/Xud.ts]: we can pass this in `handshakeData` at `pool.init` IMO. but you can leave as is, and i'll take care of this in #756.

File: test/unit/Semver.spec.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import semver from 'semver';
+import { expect } from 'chai';
+
+describe('Semver', () => {
```
[the remark this answers is not in the document]
Comment (rsercano) [reply test/unit/Semver.spec.ts]: I agree, but as you said it doesn't hurt, at some point if they decide to change their own code, we would be ready.

File: lib/p2p/Peer.ts [hunk 3]
```diff
@@ -79,6 +80,10 @@ class Peer extends EventEmitter {
   /** Connection retries max period. */
   private static readonly CONNECTION_RETRIES_MAX_PERIOD = 604800000;
 
+  public get version(): string {
```
[the remark this answers is not in the document]
Comment (rsercano) [reply lib/p2p/Peer.ts]: Changing it as private, but unfortunately as a side effect of using a third party lib here, it requires strictly string.

File: test/integration/Pool.spec.ts [hunk 4]
```diff
@@ -49,7 +49,7 @@ describe('P2P Pool Tests', async () => {
     db = new DB(loggers.db);
     await db.init();
 
-    pool = new Pool(config.p2p, loggers.p2p, db.models);
+    pool = new Pool(config.p2p, loggers.p2p, db.models, '');
```
[the remark this answers is not in the document]
Comment (rsercano) [reply test/integration/Pool.spec.ts]: I've completely removed it as per your review Daniel, now we're reading minCompatibleVersion in `Peer.ts`

**Comment by moshababo** [comment]
I'm merging this to start solving conflicts with other branches. @sangaman if you have more comments, we can have a followup. 

@rsercano thanks for the work!

**Pull request merged**